src/mbgl/style/expression/at.cpp

```diff
@@ -0,0 +1,63 @@
+#include <mbgl/style/expression/at.hpp>
+
+
+namespace mbgl {
+namespace style {
+namespace expression {
+
+EvaluationResult At::evaluate(const EvaluationContext& params) const {
+    const EvaluationResult evaluatedIndex = index->evaluate(params);
+    const EvaluationResult evaluatedInput = input->evaluate(params);
+    if (!evaluatedIndex) {
+        return evaluatedIndex.error();
+    }
+    if (!evaluatedInput) {
+        return evaluatedInput.error();
+    }
+    
+    const auto i = *fromExpressionValue<double>(*evaluatedIndex);
+    const auto inputArray = *fromExpressionValue<std::vector<Value>>(*evaluatedInput);
```

Comment:
Is it possible to convert just the indexed value, rather than the whole array?

Reply:
fromExpressionValue<T>(v), where T is member of `expression::Value`, is a no-op conversion, equivalent to v->get<T>().  I'll rewrite using the latter, since it's clearer.